boards/cats_rev1Pro/Core/Src/control/orientation_kf.c

```diff
@@ -162,20 +163,85 @@ void normalize_q(float32_t* input) {
+  } else {
+    quat_meas_acc[0] = sqrtf((scaled_imu[2] + 1.0f) / 2.0f);
+    quat_meas_acc[1] = -scaled_imu[1] / sqrtf(2.0f * (scaled_imu[2] + 1.0f));
+    quat_meas_acc[2] = scaled_imu[0] / sqrtf(2.0f * (scaled_imu[2] + 1.0f));
+  }
+
+  /* Create Quaternion Structs for IMU */
+  arm_matrix_instance_f32 quat_imu_mat;
+  arm_mat_init_f32(&quat_imu_mat, 4, 1, quat_meas_acc);
+  /* Conjugate to turn the magnetometer in the right direction */
+  arm_matrix_instance_f32 quat_imu_mat_conj;
+  float32_t quat_meas_acc_conj[4] = {quat_meas_acc[0], -quat_meas_acc[1], -quat_meas_acc[2], -quat_meas_acc[3]};
+  arm_mat_init_f32(&quat_imu_mat_conj, 4, 1, quat_meas_acc_conj);
+
+  /* Compute Quaternion from Magnetometer Data */
+
+  /* Normalize and rotate quaternion */
+  float32_t quat_meas_magneto[4] = {0, 0, 0, 0};
+  magneto_data_t processed_magno = *magneto_data;
+  float32_t ampl = sqrtf(processed_magno.magneto_x * processed_magno.magneto_x +
+                         processed_magno.magneto_y * processed_magno.magneto_y +
+                         processed_magno.magneto_z * processed_magno.magneto_z);
+  processed_magno.magneto_x /= ampl;
+  processed_magno.magneto_y /= ampl;
+  processed_magno.magneto_z /= ampl;
+  float32_t magneto_scaled[4] = {0, processed_magno.magneto_x, processed_magno.magneto_y, processed_magno.magneto_z};
+  arm_matrix_instance_f32 scaled_magneto_mat;
+  arm_mat_init_f32(&scaled_magneto_mat, 4, 1, magneto_scaled);
+
+  float32_t holder[4] = {0};
+  arm_matrix_instance_f32 holder_mat;
+  arm_mat_init_f32(&holder_mat, 4, 1, holder);
+  float32_t rotated_magneto[4] = {0};
+  arm_matrix_instance_f32 rotated_magneto_mat;
+  arm_mat_init_f32(&rotated_magneto_mat, 4, 1, rotated_magneto);
+  /* l = q_star*m*q */
+  quaternion_mat(&quat_imu_mat_conj, &scaled_magneto_mat, &holder_mat);
+  quaternion_mat(&holder_mat, &quat_imu_mat, &rotated_magneto_mat);
+  /*
+    log_trace("%ld; %ld; %ld; %ld", (int32_t)((float)rotated_magneto[0] * 1000),
+              (int32_t)((float)rotated_magneto[1] * 1000), (int32_t)((float)rotated_magneto[2] * 1000),
+              (int32_t)((float)rotated_magneto[3] * 1000));
+  */
+  float32_t tau = rotated_magneto[1] * rotated_magneto[1] + rotated_magneto[2] * rotated_magneto[2];
+  float32_t sqrt_tau = sqrtf(tau);
+  if (rotated_magneto[1] < 0) {
+    quat_meas_magneto[0] = rotated_magneto[2] / (sqrtf(2.0f) * sqrtf(tau - rotated_magneto[1] * sqrt_tau));
+    quat_meas_magneto[3] = sqrtf(tau - rotated_magneto[1] * sqrt_tau) / (sqrtf(2.0f) * sqrt_tau);
+  } else {
+    quat_meas_magneto[0] = sqrtf(tau + rotated_magneto[1] * sqrt_tau) / (sqrtf(2.0f) * sqrt_tau);
+    quat_meas_magneto[3] = rotated_magneto[2] / (sqrtf(2.0f) * sqrtf(tau + rotated_magneto[1] * sqrt_tau));
+  }
+
+  arm_matrix_instance_f32 quat_magneto_mat;
+  arm_mat_init_f32(&quat_magneto_mat, 4, 1, quat_meas_magneto);
+
+  /* Compute Quaternion */
+  float32_t quat_meas[4] = {0, 0, 0, 0};
```

Comment:
at lines 210 & 213 you're initializing with {0}. That should also be enough here.